packages/grafana-data/src/types/logs.ts

```diff
@@ -260,3 +260,37 @@ export const hasLogsContextUiSupport = (datasource: unknown): datasource is Data
 
   return withLogsSupport.getLogRowContextUi !== undefined;
 };
+
+export interface AnalyzeQueryOptions {
+  check: string;
```

Comment:
Could we make this more strict by `check: 'HAS_FILTER'` or enum to make sure this is used for its purpose?  In the future, we can always add more options.

Reply:
For now I would prefer to go with the same approach as `modifyQuery` for consistency. I documented the actions in the new interface. Let me know what you think.

Reply:
Now when we use this for toggle label filters, could we type it more strongly with `HAS_FILTER`?

Reply:
Strongly typed now.